rio_tiler/io/stac.py

```diff
@@ -148,7 +149,7 @@ class STACReader(MultiBaseReader):
     """
 
     filepath: str = attr.ib()
-    item: Dict = attr.ib(default=None)
+    item: pystac.Item = attr.ib(default=None)
```

Comment:
Could we maybe use attr.converter (https://www.attrs.org/en/stable/init.html#converters) to still allow user to pass dict?

Reply:
I could but then we cannot ensure to have the self href of the item if it is not specified as a proper "self" link in the item. this self href allows to resolve relative asset uri.

Reply:
I replaced the constructure using dict. The tiny drawback is that a dictionary without proper self is unlocatable and thus relative href shall fail.

Reply:
I think you could do 

```python
def to_stac_item(item: Union[Dict, pystac.Item]) -> pystac.Item:
    if isinstance(item, Dict):
        return pystac.Item.from_dict(item)
    return item

item: pystac.Item = attr.ib(default=None, converter=to_stac_item)
```

Reply:
ok thank you I will do that. I am not very fond of python...